### PR TITLE
doc: Tooltip boxes for Rule Unit DSL

### PR DESCRIPTION
**Pull request by tarilabs**
Adding tooltip boxes in doc for Rule Unit DSL following architectural conversations.

**JIRA**: none

**referenced Pull Requests**: none

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

- for <b>pull request checks</b>  
  Please add comment: <b>Jenkins retest this</b>

- for a <b>specific pull request check</b>  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] tests</b>

- for a <b>full downstream build</b> 
  - for <b>jenkins</b> job: please add comment: <b>Jenkins run fdb</b>
  - for <b>github actions</b> job: add the label `run_fdb`

- <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

- <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

- <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>

- for <b>quarkus branch checks</b>  
  Run checks against Quarkus current used branch  
  Please add comment: <b>Jenkins run quarkus-branch</b>

- for a <b>quarkus branch specific check</b>  
  Run checks against Quarkus current used branch  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] quarkus-branch</b>

- for <b>quarkus main checks</b>  
  Run checks against Quarkus main branch  
  Please add comment: <b>Jenkins run quarkus-main</b>

- for a <b>specific quarkus main check</b>  
  Run checks against Quarkus main branch  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] quarkus-main</b>

- for <b>quarkus lts checks</b>  
  Run checks against Quarkus lts branch  
  Please add comment: <b>Jenkins run quarkus-lts</b>

- for a <b>specific quarkus lts check</b>  
  Run checks against Quarkus lts branch  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] quarkus-lts</b>

- for <b>native checks</b>  
  Run native checks  
  Please add comment: <b>Jenkins run native</b>

- for a <b>specific native check</b>  
  Run native checks 
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] native</b>

 - for <b>native lts checks</b>  
  Run native checks against quarkus lts branch
  Please add comment: <b>Jenkins run native-lts</b>

- for a <b>specific native lts check</b>  
  Run native checks against quarkus lts branch
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] native-lts</b>
</details>

<details>
<summary>
How to backport a pull request to a different branch?
</summary>

In order to automatically create a **backporting pull request** please add one or more labels having the following format `backport-<branch-name>`, where `<branch-name>` is the name of the branch where the pull request must be backported to (e.g., `backport-7.67.x` to backport the original PR to the `7.67.x` branch).

> **NOTE**: **backporting** is an action aiming to move a change (usually a commit) from a branch (usually the main one) to another one, which is generally referring to a still maintained release branch. Keeping it simple: it is about to move a specific change or a set of them from one branch to another.

Once the original pull request is successfully merged, the automated action will create one backporting pull request per each label (with the previous format) that has been added.

If something goes wrong, the author will be notified and at this point a manual backporting is needed.

> **NOTE**: this automated backporting is triggered whenever a pull request on `main` branch is labeled or closed, but both conditions must be satisfied to get the new PR created.
</details>

### CODE REVIEW

File: drools-docs/src/modules/ROOT/pages/KIE/BuildDeployUtilizeAndRun/_rule-unit-dsl.adoc [hunk 1]
```diff
@@ -2,7 +2,17 @@
...
-In addition to the standard Rule Unit API, {PRODUCT} 8 offers an alternative way of writing rules in combination with Rule Unit. You can now define the rules for Rule Units using a dedicated set of Java APIs. Let's learn with comments on the example.
+In addition to the standard Rule Unit API, {PRODUCT} 8 offers an alternative way of writing rules in combination with Rule Unit.
+You can now define the rules for Rule Units using a dedicated set of Java APIs.
+
+TIP: As an end-user, you can think of the Rule Unit DSL as an alternative to DRL, in order to define your rules with an internal DSL using Java.
+
+CAUTION: The Rule Unit DSL is not intended to be used a way to code-generate executable model code, nor to produce DRL files.
+The goal of the Rule Unit DSL is to programmatically invoke the exec model methods (at runtime).
+As a developer using {PRODUCT} for your application development, you may find helpful to leverage _programmatically_ the Rule Unit DSL only for very dynamic use-cases
+(when the rule is only known at runtime, and compile-time performance optimizations are not needed).
```
Comment (tkobayas) [inline drools-docs/src/modules/ROOT/pages/KIE/BuildDeployUtilizeAndRun/_rule-unit-dsl.adoc]: I agree to caution `compile-time performance optimizations are not needed`, but I guess even if `the rule is only known at runtime` is not the case, some users may still prefer writing rules with Java APIs (Rule Unit DSL).

**Review by tkobayas — APPROVED** [review]
Thanks! Basically I agree your points. Just commented my personal thought (you may ignore it).

File: drools-docs/src/modules/ROOT/pages/KIE/BuildDeployUtilizeAndRun/_rule-unit-dsl.adoc (hunk 1 above)
Comment (tarilabs) [reply drools-docs/src/modules/ROOT/pages/KIE/BuildDeployUtilizeAndRun/_rule-unit-dsl.adoc]: Thank you for the feedback!

Indeed, but as you say:

> some users may still prefer writing rules with Java APIs (Rule Unit DSL).

1. This section refers to _developer_
2. _programmatic_ use is referred explicitly

which is another use-case from user writing Java code and this internal DSL.

I think I need to ponder on a way to further highlight the difference 🤔

Comment (tkobayas) [reply drools-docs/src/modules/ROOT/pages/KIE/BuildDeployUtilizeAndRun/_rule-unit-dsl.adoc]: I meant user=developer. And not imagining another use case. So basically saying the same as what you wrote.

Just I thought that even if the rule is known beforehand, user=developer may want to use Rule Unit DSL.

But the note is okay as-is, it tells enough information. Thanks!

Comment (tarilabs) [reply drools-docs/src/modules/ROOT/pages/KIE/BuildDeployUtilizeAndRun/_rule-unit-dsl.adoc]: Worthy to be indeed worked a bit more I think, thanks for highlighting!

To me there are 2 personas at play here.

The user (which would by default write DRL, can indeed opt for this internal DSL in Java "Rule Unit DSL").
The developer, like we are for the ansible integration, we use Drools and this DSL in another way ("programmatically" at runtime). 

I want to highlight clearly that if you are in the second case, you might be missing out from compile-time optimizations.

Comment (tkobayas) [reply drools-docs/src/modules/ROOT/pages/KIE/BuildDeployUtilizeAndRun/_rule-unit-dsl.adoc]: Thanks, that makes sense. I didn't think about the second (developer) case much.

**Review by mariofusco — APPROVED** [review]

File: drools-docs/src/modules/ROOT/pages/KIE/BuildDeployUtilizeAndRun/_rule-unit-dsl.adoc (hunk 1 above)
Comment (tarilabs) [reply drools-docs/src/modules/ROOT/pages/KIE/BuildDeployUtilizeAndRun/_rule-unit-dsl.adoc]: I've added fed0af1 in the hope it maybe clarifies better; what do you think?

Comment (tkobayas) [reply drools-docs/src/modules/ROOT/pages/KIE/BuildDeployUtilizeAndRun/_rule-unit-dsl.adoc]: @tarilabs Yes, it's completely clear to me. Thank you!

**Pull request merged**